definitions/npm/react-redux_v5.x.x/flow_v0.89.x-/test_connect.js

```diff
@@ -0,0 +1,495 @@
+  };
+
+  const mapStateToProps = () => ({});
+  const mapDispatchToProps = () => ({});
+
+  const mergeProps = () => ({
+    // $ExpectError wrong type for stringProp
+    stringProp: true
+  });
+
+  connect(mapStateToProps, mapDispatchToProps, mergeProps)(Component);
+}
+
+function testExactProps() {
+  type Props = {|
+    forMapStateToProps: string,
+    passthrough: number,
+    fromStateToProps: string
+  |};
+
+  class Com extends React.Component<Props> {
+    render() {
+      return <div>{this.props.passthrough} {this.props.fromStateToProps}</div>;
+    }
+  }
+
+  type State = {a: number};
+  type InputProps = {|
+    forMapStateToProps: string,
+    passthrough: number,
+  |};
+
+  const mapStateToProps = (state: State, props: InputProps) => {
+    return {
+      fromStateToProps: 'str' + state.a
+    }
+  };
+
+  const Connected = connect(mapStateToProps)(Com);
+  <Connected passthrough={123} forMapStateToProps={'data'} />;
+  //$ExpectError extra prop what exact props does not allow
+  <Connected passthrough={123} forMapStateToProps={321} extraProp={123}/>;
+  //$ExpectError wrong type for forMapStateToProps
+  <Connected passthrough={123} forMapStateToProps={321}/>;
+  //$ExpectError passthrough missing
+  <Connected forMapStateToProps={'data'} />;
+  //$ExpectError forMapStateToProps missing
+  <Connected passthrough={123}/>;
+  //$ExpectError takes in only React components
+  connect(mapStateToProps)('');
+}
+
+function testWithStatelessFunctionalComponent() {
+  type Props = {passthrough: number, fromStateToProps: string};
+  const Com = (props: Props) => <div>{props.passthrough} {props.fromStateToProps}</div>
+
+  type State = {a: number};
+  type InputProps = {
+    forMapStateToProps: string
+  };
+  const mapStateToProps = (state: State, props: InputProps) => {
+    return {
+      fromStateToProps: 'str' + state.a
+    }
+  };
+
+  const Connected = connect(mapStateToProps)(Com);
+  <Connected passthrough={123} forMapStateToProps={'data'}/>;
+  //$ExpectError wrong type for passthrough
+  <Connected passthrough={''} forMapStateToProps={'data'}/>;
+  //$ExpectError wrong type for forMapStateToProps
+  <Connected passthrough={123} forMapStateToProps={321} />;
+  //$ExpectError passthrough missing
+  <Connected forMapStateToProps={'data'} />;
+  //$ExpectError forMapStateToProps missing
+  <Connected passthrough={123}/>;
+  //$ExpectError takes in only React components
+  connect(mapStateToProps)('');
+}
+
+function testMapStateToPropsDoesNotNeedProps() {
+  type Props = {passthrough: number, fromStateToProps: string};
+  class Com extends React.Component<Props> {
+    render() {
+      return <div>{this.props.passthrough}</div>;
+    }
+  }
+
+  type State = {a: string}
+  const mapStateToProps = (state: State) => {
+    return {
+      fromStateToProps: state.a
+    }
+  }
+
+  const Connected = connect(mapStateToProps)(Com);
+  <Connected passthrough={123}/>;
+  //$ExpectError component property passthrough not found
```

Comment:
getting 

```
    
    Error -------------------------------------------------------------------------------------------- test_connect.js:227:3
    
    Cannot create `Connected` element because property `fromStateToProps` is missing in props [1] but exists in `Props` [2].
    
       test_connect.js:227:3
       227|   <Connected />;
              ^^^^^^^^^^^^^ [1]
    
    References:
       test_connect.js:211:37
       211|   class Com extends React.Component<Props> {
                                                ^^^^^ [2]
    
    
    
    Found 1 error
    
    Only showing the most relevant union/intersection branches.
    To see all branches, re-run Flow with --show-all-branches
```

instead of missing `passthrough`.